format_spec.md

```diff
@@ -0,0 +1,339 @@
+
+##### `icon` (optional)
+
+This is an object that contains the following fields:
+
+##### `sha256` (optional)
+
+An optional SHA256 checksum of the icon specified in `uri`.
+
+##### `uri`
+
+The URI of this icon. This field is required to be present if the `icon` object is present.
+
+---
+
+### `license` (optional)
+
+This is an object that contains information about the license that a project is under. It contains the following fields:
+
+##### `spdx`
+
+The SPDX identifier of the license. [You can find a list of SPDX license names and identifiers here](https://spdx.org/licenses/).
+
+##### `uri`
+
+A URL linking to the full license.
+
+##### `modpackUsage` (optional)
+
+A field describing usage of the Project in modpacks. Valid options:
+
+- `alwaysAllow` - This Project can be used in a modpack without asking for permission.
+- `requiresPermission` - Before included in a modpack, permission must be granted from the author.
+- `neverAllow` - This Project can never be used in a modpack.
+
+---
+
+### `links` (optional)
+
+This field MUST be stored as an array. It contains a collection of links relating to the Project. Each Link Object stored inside this arrow MUST contain the following fields:
+
+##### `rel`
+
+This field MUST be stored as an array OR a String. If stored as an array, it MUST contain Strings. This field contains information about what the link is. Valid standardized values are:
+
+- `homepage` - The official homepage of a Project
+- `documentation` - Documentation of a Project.
+- `wiki` - Wiki information about a Project.
+- `scm` - The source code of a Project, for example a GitHub URL.
+- `issues` - An issue tracker where bugs and feature requests can be filed.
+- `community` - A forum or other online discussion platform where users can discuss the Project.
+
+Other values MAY be used, such as `bitbucket` or `github`. However, it is RECOMMENDED to use generic terminology instead of specific names. If you are unsure whether a value COULD become standardized, please prefix it with `x-`.
+
+##### `name`
+
+A human-readable name of the link. For example, "Source Code" or "Project Website".
+
+##### `uri`
+
+The link, to be viewed in a web browser.
+
+---
+
+### `successor` and `predecessor` (optional)
+
+If an older or newer version of a Project exists, but has a separate ID, these fields may be used. These fields MUST be either an array or String representing the respective projects' IDs. If the field is an array, the Projects should be listed in chronological order from when they were first released.
+
+```
+"successor": [
+  "first-successor",
+  "second-successor
+]
+```
+
+In the example above, the project `first-successor` was released first. After the release of `first-successor`, the project `second-successor` was released.
+
+---
+
+## Version Information
+
+Projects can contain a `versions` array, which lists Version Objects.
```

Comment:
Does the order matter? Especially since some projects use version numbers that aren't machine-comparable and some may reset version numbers or change numbering schemes. Newest to oldest or oldest to newest?